packages/manager/apps/pci-billing/src/components/consumption/SavingsPlanList.component.tsx

```diff
@@ -0,0 +1,113 @@
+import { OsdsSpinner, OsdsText } from '@ovhcloud/ods-components/react';
+import {
+  ODS_SPINNER_SIZE,
+  ODS_TEXT_LEVEL,
+  ODS_TEXT_SIZE,
+} from '@ovhcloud/ods-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { Currency } from '@ovh-ux/manager-config';
+import { TFunction } from 'i18next';
+import {
+  Datagrid,
+  DataGridTextCell,
+  useDataGrid,
+} from '@ovh-ux/manager-react-components';
+import { useContext } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { TSavingsPlan } from '@/api/hook/useConsumption';
+import { useSavingsPlanDetails } from '@/hooks/useSavingsPlanDetails';
+import { useServiceId } from '@/hooks/useServiceId';
+
+const getColumns = ({
+  t,
+  currency,
+}: {
+  currency: Currency;
+  t: TFunction;
+}): {
+  id: keyof (TSavingsPlan & { displayName: string });
+  label: string;
+  cell: (row: TSavingsPlan & { displayName: string }) => JSX.Element;
+}[] => [
+  {
+    id: 'id',
+    cell: (row) => (
+      <DataGridTextCell>{row?.displayName || row?.id}</DataGridTextCell>
+    ),
+    label: t('cpbc_savings_plan_col_name'),
+  },
+  {
+    id: 'size',
+    cell: (row) => <DataGridTextCell>{row.size}</DataGridTextCell>,
+    label: t('cpbc_savings_plan_col_size'),
+  },
+  {
+    id: 'flavor',
+    cell: (row) => <DataGridTextCell>{row.flavor}</DataGridTextCell>,
+    label: t('cpbc_savings_plan_col_flavor'),
+  },
+  {
+    id: 'totalPrice',
+    cell: (row) => (
+      <DataGridTextCell>{`${(row.totalPrice || 0).toFixed(2)} ${
+        currency.symbol
+      }`}</DataGridTextCell>
+    ),
+    label: t('cpbc_savings_plan_col_price'),
+  },
+];
+
+const SavingsPlanList = ({
+  monthlySavingsPlanList,
+}: {
+  monthlySavingsPlanList: TSavingsPlan[];
+}) => {
+  const { pagination, setPagination } = useDataGrid();
+
+  const { t } = useTranslation('consumption/monthly-instance');
+  const { currency } = useContext(ShellContext).environment.getUser();
+  const { projectId } = useParams<{ projectId: string }>();
+  const { data: serviceId } = useServiceId(projectId || '');
+  const { enrichedSavingsPlans, isLoading } = useSavingsPlanDetails(
+    serviceId?.toString() || '',
+    monthlySavingsPlanList,
+  );
+
+  const columns = getColumns({ t, currency });
```

Comment:
a useMemo could be great since there are only 2 parameters

Or else it should be put inside the call to the Datagrid